controllers/etcdcluster_controller.go

```diff
@@ -0,0 +1,82 @@
+package controllers
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	"time"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
+)
+
+// EtcdClusterReconciler reconciles a EtcdCluster object
+type EtcdClusterReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters/status,verbs=get;update;patch
+
+func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	log := r.Log.WithValues("etcdcluster", req.NamespacedName)
+
+	var cluster etcdv1alpha1.EtcdCluster
+	if err := r.Get(ctx, req.NamespacedName, &cluster); err != nil {
+		log.Error(err, "unable to fetch EtcdCluster")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	// your logic here
+	service := &v1.Service{
```

Comment:
Can we at least implement a `Get` of this resource first, and not call `Create` if a resource with the given name exists? As it stands, this code will fairly quickly go into an error loop.

Reply:
Without the `Owns(&corev1.Service{}).` it won't actually crash-loop I don't think? 😂 But I take the point. We were going to do that part next but we can do it here if you'd prefer.